### PR TITLE
Fix JsonPath.TryParse to allow relative start, consistent with JsonPa…

### LINKED ISSUE #797: JsonPath's PathParsingOptions.AllowRelativePathStart is only respected in Parse - not TryParse
### Nuget Package

json-everything.net (playground)

### Package Version

1.1.5

### Operating System

Windows

### .Net Target (if relevant, please specify the version in the description)

.Net (5 or after)

### A clear and concise description of the bug.

There is a discrepancy with how `JsonPath.Parse` and `JsonPath.TryParse` handle the `PathParsingOptions.AllowRelativePathStart` property:

- `JsonPath.Parse` uses `!options.AllowRelativePathStart` as the value for `PathParser.Parse`'s `requireGlobal` parameter.  ([Source](https://github.com/json-everything/json-everything/blob/96e1ccfe6e233199ece276b7bc7328b1bfeac07e/src/JsonPath/JsonPath.cs#L74))
- `JsonPath.TryParse` always passes `true` as the value for `PathParser.TryParse`'s `requireGlobal` parameter.  ([Source](https://github.com/json-everything/json-everything/blob/96e1ccfe6e233199ece276b7bc7328b1bfeac07e/src/JsonPath/JsonPath.cs#L102))

### What did you expect?

I expect `Parse` and `TryParse` to work the same, with the exception that `TryParse` doesn't throw an exception.

### Please add test code or steps to reproduce the behavior.

```csharp
Assert.True(
    JsonPath.TryParse(
        "@.anything", 
        out _, 
        new PathParsingOptions() 
        { 
            AllowRelativePathStart = true 
        }
    )
);
```

### Is there any other information you'd like to share regarding this bug?

_No response_

### Code of Conduct & Contributor Guidelines

- [X] I agree to follow this project's Code of Conduct and Contribution Guidelines.

### PR DESCRIPTION
**Pull request by mikechristiansenvae**
### Description

There is a discrepancy with how JsonPath.Parse and JsonPath.TryParse handle the PathParsingOptions.AllowRelativePathStart property:

- JsonPath.Parse uses !options.AllowRelativePathStart as the value for PathParser.Parse's requireGlobal parameter. ([Source](https://github.com/json-everything/json-everything/blob/96e1ccfe6e233199ece276b7bc7328b1bfeac07e/src/JsonPath/JsonPath.cs#L74))
- JsonPath.TryParse always passes true as the value for PathParser.TryParse's requireGlobal parameter. ([Source](https://github.com/json-everything/json-everything/blob/96e1ccfe6e233199ece276b7bc7328b1bfeac07e/src/JsonPath/JsonPath.cs#L102))

This MR fixes the discrepancy by using the same logic in both methods.

### Links

Resolves #797 

### Checks

- [X] I have read and understand the [Code of Conduct](https://github.com/json-everything/json-everything/blob/master/CODE_OF_CONDUCT.md) and [Contribution Guidelines](https://github.com/json-everything/json-everything/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by gregsdennis — CHANGES_REQUESTED** [review]
Thanks @mikechristiansenvae.  Would you add in a few tests, please?

**Comment by mikechristiansenvae** [comment]
@gregsdennis - Tests added.

File: src/JsonPath.Tests/ParsingTests.cs [hunk 1]
```diff
@@ -139,4 +139,19 @@ public void ParseInOpWithoutOptions(string path)
...
 	}
+
+
+
+	[TestCaseSource(nameof(SuccessCases))]
+	public void ParseRelativeStarts(string path)
+	{
+		path = $"@{path[1..]}"; // Turn the absolute path into a relative path
+		Assert.DoesNotThrow(() => JsonPath.Parse(path, new PathParsingOptions(){AllowRelativePathStart = true}));
+	}
```
Comment (2BitSalute) [inline src/JsonPath.Tests/ParsingTests.cs:150]: Nit: there should be one line between adjacent tests/methods (here, you have none, and above, you have 3)

Comment (mikechristiansenvae) [reply src/JsonPath.Tests/ParsingTests.cs:148]: Sorry, fixed!

File: src/JsonPath.Tests/ParsingTests.cs [hunk 2]
```diff
@@ -139,4 +139,18 @@ public void ParseInOpWithoutOptions(string path)
...
+
+	[TestCaseSource(nameof(SuccessCases))]
+	public void TryParseRelativeStarts(string path)
```
Comment (gregsdennis) [inline src/JsonPath.Tests/ParsingTests.cs:151]: I suppose I need to throw `.TryParse()` tests in here now :)

I have the test suite set up to run `.TryParse()` and then `.Parse()`, but it'd be good to do it here, too.

**Review by gregsdennis — APPROVED** [review]

**Pull request merged**